clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -344,16 +347,26 @@ static void canonicalizeIntrinsicThreeWayCmp(CIRBaseBuilderTy &builder,
 void LoweringPreparePass::lowerVAArgOp(VAArgOp op) {
   CIRBaseBuilderTy builder(getContext());
   builder.setInsertionPoint(op);
-  cir::CIRDataLayout datalayout(theModule);
 
-  auto res = cxxABI->lowerVAArg(builder, op, datalayout);
+  auto res = cxxABI->lowerVAArg(builder, op, *datalayout);
   if (res) {
     op.replaceAllUsesWith(res);
     op.erase();
   }
   return;
 }
 
+void LoweringPreparePass::lowerDeleteArrayOp(DeleteArrayOp op) {
```

Comment:
I looked at it the PR again and got something I didn't before: `lowerDeleteArrayOp` is introduced but is never called, therefore we never hit the assert you added in `lowerDeleteArray` - please make sure you mention that there are NFC parts attached to a PR because they are tricky to spot.

This approach basically changes the behavior in a way that if you stop working on this, it will be really hard for a newcomer to spot what's missing and continue the work. I suggest instead to already call `lowerDeleteArrayOp` in LoweringPrepare and change `lowerDeleteArray` to instead of having an unreacheable, have a assert on a missing feature you're going to also add as part of this PR.

Reply:
That sounds fine, just to double check: this will end up hitting an assert instead of silently not emitting anything, right? If so LGTM

Reply:
Yes